src/utils/AsyncEventEmitter.ts

```diff
@@ -180,4 +207,15 @@ export class AsyncEventEmitter<
     this.readyState = AsyncEventEmitterReadyState.DEACTIVATED
     log('set state to:', this.readyState)
   }
+
+  private removeInternalListeners<Event extends keyof EventMap>(
+    event?: Event
+  ): void {
+    if (event) {
+      this._events[event] = []
```

Comment:
Missing `return` statement: passing an explicit event name will still clear _all the events because this condition doesn't short-circuit.